boards/arm/cc1352p1_launchxl/board_antenna.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2021 Florin Stancu
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Implements the RF driver callback to configure the on-board antenna
+ * switch.
+ */
+
+#include <zephyr/init.h>
+
+#include <ti/drivers/rf/RF.h>
+#include <driverlib/gpio.h>
+#include <driverlib/ioc.h>
+#include <driverlib/rom.h>
+
+/* DIOs for RF antenna paths */
+#define BOARD_RF_24GHZ     28
+#define BOARD_RF_HIGH_PA   29
+#define BOARD_RF_SUB1GHZ   30
+
+
+static void board_cc13xx_rf_callback(RF_Handle client, RF_GlobalEvent events,
+		void *arg);
+
+
+const RFCC26XX_HWAttrsV2 RFCC26XX_hwAttrs = {
+	.hwiPriority        = INT_PRI_LEVEL7,
+	.swiPriority        = 0,
+	.xoscHfAlwaysNeeded = true,
+	/* RF driver callback for custom antenna switching */
+	.globalCallback = board_cc13xx_rf_callback,
+	/* Subscribe to events */
+	.globalEventMask = (RF_GlobalEventRadioSetup |
+			RF_GlobalEventRadioPowerDown),
+};
+
+/**
+ * Antenna switch GPIO init routine.
+ */
+static int board_antenna_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	/* set all paths to low */
+	IOCPinTypeGpioOutput(BOARD_RF_24GHZ);
+	GPIO_setOutputEnableDio(BOARD_RF_24GHZ, GPIO_OUTPUT_ENABLE);
+	IOCPinTypeGpioOutput(BOARD_RF_HIGH_PA);
+	GPIO_setOutputEnableDio(BOARD_RF_HIGH_PA, GPIO_OUTPUT_ENABLE);
+	IOCPinTypeGpioOutput(BOARD_RF_SUB1GHZ);
+	GPIO_setOutputEnableDio(BOARD_RF_SUB1GHZ, GPIO_OUTPUT_ENABLE);
```

Comment:
please use GPIO API and Devicetree to specify antenna settings. -1 on using HAL in tree for configuring GPIOs.

Reply:
Hello, I actually did that in my last commit: https://github.com/zephyrproject-rtos/zephyr/pull/46050/commits/ceaf79b53a06a6e90a057ac060f0b765fd47ee43

Reply:
please squash changes

Reply:
Alright, I will do a rebase a bit later today and re-test on my boards. Thanks.